app/test_split_integer.py

```diff
@@ -2,20 +2,21 @@
 
 
 def test_sum_of_the_parts_should_be_equal_to_value() -> None:
-    pass
+    assert sum(split_integer(17, 4)) == 17
 
 
 def test_should_split_into_equal_parts_when_value_divisible_by_parts() -> None:
-    pass
+    values = split_integer(20, 4)
+    assert all(map(lambda x: x == values[0], values))
 
 
 def test_should_return_part_equals_to_value_when_split_into_one_part() -> None:
-    pass
+    assert split_integer(5, 1) == [5]
 
 
 def test_parts_should_be_sorted_when_they_are_not_equal() -> None:
-    pass
+    assert split_integer(32, 6) == sorted(split_integer(32, 6))
 
 
 def test_should_add_zeros_when_value_is_less_than_number_of_parts() -> None:
-    pass
+    assert split_integer(3, 4) == [0, 1, 1, 1]
```

Comment:
The expected result `[0, 1, 1, 1]` does not satisfy the condition that the difference between the max and min number in the array should be <= 1. According to the task description, the array should be sorted in ascending order and the difference between the max and min should be <= 1. Please revise the expected result to meet these criteria.